build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/docker/DockerBuildTask.java

```diff
@@ -182,9 +183,9 @@ public void execute() {
             });
 
             try {
-                parameters.getMarkerFile().getAsFile().get().createNewFile();
+                Files.writeString(parameters.getMarkerFile().getAsFile().get().toPath(), String.valueOf(System.currentTimeMillis()));
```

Comment:
This has been changed so that the `cloud-ess` image will rebuild when the `cloud` image changes.

Reply:
That seems wrong. That implies that the tasks that build both images produce the same output marker file. We should change that so they have distinct outputs instead.

Reply:
I don't follow. Each `DockerBuildTask` task creates its own marker file. The issue was that it never changed.

Reply:
I guess my next question is, why should it have to? The task should run again if and only if a) the inputs change or b) the output (in this case the marker file) is deleted or modified. The content of the output file should be irrelevant as it's perfectly fine for change inputs to make a task run again and produce the same output.

If updating something didn't trigger the task to run again, that means we're missing an input somewhere, not that the output is wrong. Or there's something else interacting here that I'm not following.

Reply:
Actually, it makes more sense to emit the image's SHA checksum into the marker file, since that shouldn't change if the docker build resulted in the same image.

Reply:
Actually redux: the way we build our images makes is hostile to building images with the same checksum. I've made another change to make the build more cache-friendly, but only for snapshot builds. This yields a repeatable checksum output for a given image.